dot/parachain/backing/per_relay_parent_state.go

```diff
@@ -189,10 +189,39 @@ type perRelayParentState struct {
 }
 
 func (rpState *perRelayParentState) coreIndexFromStatement(
-	parachaintypes.SignedFullStatementWithPVD,
-) (parachaintypes.CoreIndex, error) {
-	// TODO: Implement this #4324
-	return parachaintypes.CoreIndex{}, nil
+	signedStatementWithPVD parachaintypes.SignedFullStatementWithPVD,
+) (*parachaintypes.CoreIndex, error) {
+	groupIndex, ok := rpState.validatorToGroup[signedStatementWithPVD.SignedFullStatement.ValidatorIndex]
+	if !ok {
+		return nil, fmt.Errorf("validator index not found in validator to group mapping")
+	}
+
+	coreIndex := rpState.groupRotationInfo.CoreForGroup(groupIndex, uint(rpState.numOfCores))
+	if coreIndex.Index > rpState.numOfCores {
+		return nil, fmt.Errorf("invalid core index %d, expected core index to be less than %d",
+			coreIndex.Index, rpState.numOfCores)
+	}
+
+	statementVDTValue, err := signedStatementWithPVD.SignedFullStatement.Payload.Value()
```

Comment:
quick question regarding the statementVDT @axaysagathiya, I noticed that the statement can only be either Valid or Seconded. Do those two values represent the first round check and secondary check of candidate?